midenc-compile/src/stages/cross_ctx/lift_exports.rs

```diff
@@ -0,0 +1,170 @@
+//! lifting the exports for cross-context calls.
+
+use std::collections::BTreeMap;
+
+use miden_assembly::Spanned;
+use midenc_hir::{
+    diagnostics::Severity,
+    pass::AnalysisManager,
+    types::Abi::{self, Canonical},
+    ComponentBuilder, ComponentExport, FunctionType, InstBuilder, InterfaceFunctionIdent,
+};
+use midenc_session::{DiagnosticsHandler, Session};
+
+use super::flat::{
+    assert_core_wasm_signature_equivalence, flatten_function_type, needs_transformation,
+    FlatteningDirection,
+};
+use crate::{stage::Stage, CompilerResult, LinkerInput};
+
+/// Generates lifting for exports for the cross-context calls according to the Miden ABI.
+///
+/// For each component export ensure a module for each interface that has a lifting
+/// function, i.e. a function that takes parameters according to the cross-context Miden
+/// ABI, converts them to the Wasm CABI, calls the core Wasm module exported function,
+/// converts the results to the cross-context Miden ABI
+///
+/// After this stage all exported functons are expected to be called using the Miden ABI for
+/// cross-context calls, i.e. using the stack and the advice provider for arguments and results.
+pub struct LiftExportsCrossCtxStage;
+
+impl Stage for LiftExportsCrossCtxStage {
```

Comment:
*Question:* I would expect us to emit IR from the Wasm frontend where the interfaces/component exports have already been materialized, i.e. when translating an exported component instance (as an interface), we would emit any synthetic exports as part of that translation. Is the current approach here just an artifact of the current IR structure, and can't really be changed until HIR2 is landed? If so, that's fine, but let's plan on moving this to the Wasm frontend as part of integrating HIR2, since it feels a bit out of place here.

Reply:
My plan is to move it to the frontend during the HIR2 integration. 
Initially, I tried to figure out how to do it in the frontend, but IIRC due to the order of the translation in the frontend, I was missing some info on component import/export during the core module parsing. This, and the intention to build a prototype, pushed me to make it a separate stage(s). But since we'll revamp Wasm CM parsing (ditch inlining, etc.) and I now have a clearer picture of how it works (thanks to the prototype) I should be able to do it in the frontend.